sdk/identity/Azure.Identity/src/VisualStudioCredential.cs

```diff
@@ -0,0 +1,242 @@
+                if (!_fileSystem.FileExists(tokenProvider.Path))
+                {
+                    continue;
+                }
+
+                arguments.Clear();
+                arguments.Append(ResourceArgumentName).Append(' ').Append(resource);
+
+                if (_tenantId != default)
+                {
+                    arguments.Append(' ').Append(TenantArgumentName).Append(' ').Append(_tenantId);
+                }
+
+                // Add the arguments set in the token provider file.
+                if (tokenProvider.Arguments?.Length > 0)
+                {
+                    foreach (var argument in tokenProvider.Arguments)
+                    {
+                        arguments.Append(' ').Append(argument);
+                    }
+                }
+
+                var startInfo = new ProcessStartInfo
+                {
+                    FileName = tokenProvider.Path,
+                    Arguments = arguments.ToString(),
+                    UseShellExecute = false,
+                    ErrorDialog = false,
+                    CreateNoWindow = true,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                };
+
+                processStartInfos.Add(startInfo);
+            }
+
+            return processStartInfos;
+        }
+
+        private VisualStudioTokenProvider[] GetTokenProviders(string tokenProviderPath)
+        {
+            var content = GetTokenProviderContent(tokenProviderPath);
+
+            JsonElement providersElement = JsonDocument.Parse(content).RootElement.GetProperty("TokenProviders");
+
+            var providers = new VisualStudioTokenProvider[providersElement.GetArrayLength()];
+            for (int i = 0; i < providers.Length; i++)
+            {
+                JsonElement providerElement = providersElement[i];
+
+                var path = GetStringPropertyValue(providerElement, "Path");
+                var arguments = GetStringArrayPropertyValue(providerElement, "Arguments");
+                var preference = GetInt32PropertyValue(providerElement, "Preference");
+
+                providers[i] = new VisualStudioTokenProvider(path, arguments, preference);
+            }
+
+            Array.Sort(providers);
+            return providers;
+        }
+
+        private string GetTokenProviderContent(string tokenProviderPath)
+        {
+            try
+            {
+                return _fileSystem.ReadAllText(tokenProviderPath);
+            }
+            catch (FileNotFoundException exception)
+            {
+                throw new CredentialUnavailableException($"Visual Studio Token provider file not found at {tokenProviderPath}", exception);
+            }
+        }
+
+        private static string GetStringPropertyValue(JsonElement element, string name) => element.GetProperty(name).GetString();
```

Comment:
There is a single usage of this method, worth inlining?

Reply:
Inlined